Assets/HoloToolkit/SpatialSound/Scripts/UAudioManager/ActiveEvent.cs

```diff
@@ -61,6 +61,12 @@ public GameObject AudioEmitter
             private set;
         }
 
+        public string MessageToSend
```

Comment:
I'm fairly new to Unity, and I believe I'm following the conversation here, but I'm go to restate it just to make sure. So please feel free to aim me in the right direction if I've got something wrong.  I used the SendMessage Event system which is baked into Unity to do this, and I understand it is quite slow because it is using reflection internally.  

From reading above I believe that @stbertou and @NeerajW are suggesting the same thing, which is to follow the pattern recently added for OnGazeEnterEvent.  The advantage of this is that I get visibility in the Unity editor and can configure the receiver of the event there instead of passing it on the play method as I currently have it implemented.  This does not result in an improvement in performance since it still uses SendMessage to decide when to trigger the event.

Here is why I don't think this is appropriate for this situation.  Imagine that I have UAudioManager configure to play 3 sounds: Sound1, Sound2, Sound3.  Now imagine that I want  to do something after Sound1 is complete but not after Sound2 and Sound3.  How I currently have this implemented I would do this with:

```
audio.PlayEvent("Sound1", gameObject, "OnSound1Complete");
audio.PlayEvent("Sound2", gameObject);
audio.PlayEvent("Sound3", gameObject);
```

Following the unity event pattern I don't see how you accomplish the same thing, you would attach an event  and it would get executed after every sound is completed. 

Does that make sense?